app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :books
+
+  has_one_attached :profile_image
+  enum :user_type, [:author, :librarian, :consumer], default: :consumer
+
+  validates :name, presence: true
+  validates :national_id, presence: true
+  validates :user_type, presence: true
+  validates :profile_image, presence: true
+
+  validate :profile_image_type
+
+  private
+
+  def profile_image_type
```

Comment:
Use expressive names for functions that are used as validators

```suggestion
  def profile_image_type_in_allowed_types
```